amazon-q/README.md

```diff
@@ -0,0 +1,143 @@
+---
+display_name: Amazon Q
+description: Run Amazon Q in your workspace to access Amazon's AI coding assistant.
+icon: ../.icons/aws.svg
+maintainer_github: coder
+verified: true
+tags: [ai, helper, amazon-q]
+---
+
+# Amazon Q
+
+Run [Amazon Q](https://aws.amazon.com/q/) in your workspace to access Amazon's AI coding assistant. This module installs and launches Amazon Q, with support for background operation, task reporting, and custom pre/post install scripts.
+
+```tf
+module "amazon-q" {
+  source   = "registry.coder.com/modules/amazon-q/coder"
+  version  = "1.0.0"
+  agent_id = coder_agent.example.id
+  # Required: see below for how to generate
+  experiment_auth_tarball = var.amazon_q_auth_tarball
+}
+```
+
+## Prerequisites
+
+- You must generate an authenticated Amazon Q tarball on another machine:
+  ```sh
+  cd ~/.local/share/amazon-q && tar -c . | zstd | base64 -w 0
+  ```
+  Paste the result into the `experiment_auth_tarball` variable.
+- To run in the background, your workspace must have `screen` or `tmux` installed.
+
+<details>
+<summary><strong>How to generate the Amazon Q auth tarball (step-by-step)</strong></summary>
+
+**1. Install and authenticate Amazon Q on your local machine:**
+
+- Download and install Amazon Q from the [official site](https://aws.amazon.com/q/developer/).
+- Run `q login` and complete the authentication process in your terminal.
+
+**2. Locate your Amazon Q config directory:**
+
+- The config is typically stored at `~/.local/share/amazon-q`.
+
+**3. Generate the tarball:**
+
+- Run the following command in your terminal (Linux/macOS):
+  ```sh
+  cd ~/.local/share/amazon-q
```

Comment:
On macOS the path is different, and I haven't tested whether a tarball from macOS will work on Linux. If we document macOS in the README, we must make sure that the instructions are correct.